### PR TITLE
Add `Headless` to Telemetry Environment.

### PR DESCRIPTION
**Pull request by chutten**
Fixes bug 1536175

Might be nice to know if the ping's coming from a headless Firefox.

Checklist for reviewer:

- [ ] Scan the PR and verify that no changes (particularly to `.circleci/config.yml`) will cause environment variables (particularly credentials) to be exposed in test logs
- [ ] Trigger the `integration` CI test by pushing this revision [as discussed in the README](https://github.com/mozilla-services/mozilla-pipeline-schemas#packaging-and-integration-tests-optional)

For glean changes:
- [ ] Update `include/glean/CHANGELOG.md`

### CODE REVIEW

File: templates/include/telemetry/environment.1.parquetmr.txt [hunk 1]
```diff
@@ -73,6 +73,7 @@ optional group environment {
       }
       optional boolean DWriteEnabled;
       optional boolean D2DEnabled;
+      optional boolean Headless;
```
Comment (mreid-moz) [inline templates/include/telemetry/environment.1.parquetmr.txt]: Drive-by: I think this change is not compatible with schema evolution in parquet (change to the fields in a nested struct).

Comment (chutten) [reply templates/include/telemetry/environment.1.parquetmr.txt]: Is there a way I can test that assumption?

(( or I could just drop it. It's only a nice-to-have ))

Comment (mreid-moz) [reply templates/include/telemetry/environment.1.parquetmr.txt]: I'm not sure if there's an easy way to test it, so +1 for dropping :-/

**Comment by chutten** [comment]
Alrighty, it's been dropped. Ready for review.

File: validation/telemetry/event.4.headless-sample.fail.json [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        "system": {
+          "model": null,
+          "revision": null
+        }
+      },
+      "gfx": {
+        "D2DEnabled": null,
+        "DWriteEnabled": null,
+        "ContentBackend": "Skia",
+        "Headless": 0,
```
Comment (acmiyaguchi) [inline validation/telemetry/event.4.headless-sample.fail.json]: Could you pare down this validation ping? One thought is to use the `main.4.min.pass` document as a base to avoid the extra boilerplate.

https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/151c1dbbac66f7c52285e32aa7edc494957e664a/validation/telemetry/main.4.min.pass.json#L1-L16

The cases themselves look good otherwise.

Comment (chutten) [reply validation/telemetry/event.4.headless-sample.fail.json]: Alrighty. It took a little finagling since introducing the optional `environment` means you need to add some required things (like almost the entirety of `environment.build`), but I think this is now the minimal example of a passing- and failing-due-to-`Headless` schema validation set.

**Comment by chutten** [comment]
Validation set is now minimal. Good for another look
